### PR TITLE
Fix duplicate artifacts issue with maven-source-plugin in release build

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

dev/release/003-release-perform.sh fails with errors
```
[INFO] [INFO] --- source:3.3.0:jar (attach-sources) @ buildtools ---
[INFO] [INFO] Building jar: /Users/lari/workspace-pulsar/bookkeeper/target/checkout/buildtools/target/buildtools-4.17.0-sources.jar
[INFO] [INFO]
[INFO] [INFO] --- source:3.3.0:jar-no-fork (attach-sources) @ buildtools ---
[INFO] [ERROR] We have duplicated artifacts attached.
...
[INFO] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-source-plugin:3.3.0:jar-no-fork (attach-sources) on project buildtools: Presumably you have configured maven-source-plugn to execute twice times in your build. You have to configure a classifier for at least on of them. -> [Help 1]
[INFO] [ERROR]
```

After resolving that, there was another set of errors in installing `bookkeeper-dist/all`, `bookkeeper-dist/bkctl` and `bookkeeper-dist/server`. The error message contained this message:
```
The packaging plugin for this project did not assign a main file to the project but it has attachments.
```

### Changes

- use `jar-no-fork` goal instead of `jar` goal.
- change parent pom version to 29 (also used in apache/pulsar)
  - [parent pom diff between 26 -> 29](https://github.com/apache/maven-apache-parent/compare/apache-26..apache-29#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8)
-  rely on pluginManagement for handling maven plugin versions for these core plugins
    - maven-assembly
    - maven-clean-plugin
    - maven-dependency-plugin
    - maven-deploy-plugin
    - maven-install-plugin
    - maven-jar-plugin
    - maven-javadoc-plugin
    - maven-shade-plugin.
    - maven-source-plugin
- let Apache parent pom version 29 assign the plugin versions so that they are compatible with each other and the Apache release.
- Changing package from jar to pom for `bookkeeper-dist/all`, `bookkeeper-dist/bkctl` and `bookkeeper-dist/server`
  - error was `The packaging plugin for this project did not assign a main file to the project but it has attachments.`

### CODE REVIEW

**Comment by lhotari** [comment]
please also review and merge #4235 and #4236 since those were necessary at least in my case (MacOS Apple Silicon M3 Max)

**Comment by lhotari** [comment]
@eolivelli @nicoloboschi This is also needed in branch-4.16 and branch-4.15 since those are also broken due to pom.xml changed made while fixing integration tests. Please also include #4235 and #4236.

**Review by eolivelli — APPROVED** [review]
+1
thank you very much!

File: bookkeeper-dist/all/pom.xml [hunk 1]
```diff
@@ -152,7 +152,13 @@
...
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-deploy-plugin</artifactId>
-        <version>${maven-deploy-plugin.version}</version>
+        <configuration>
+          <skip>true</skip>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-install-plugin</artifactId>
```
Comment (eolivelli) [inline bookkeeper-dist/all/pom.xml:161]: This wounds weird

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: bookkeeper-dist/all/pom.xml (hunk 1 above)
Comment (lhotari) [reply bookkeeper-dist/all/pom.xml]: Thanks for the suggestions on Slack @eolivelli . I have resolved this by reverting the changes to skip install and instead changed packaging from jar to pom for `bookkeeper-dist/all`, `bookkeeper-dist/bkctl` and `bookkeeper-dist/server`.

**Review by nicoloboschi — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Pull request merged**